### ISSUE TITLE
Fix S2699 FP: When using NSubstitute Received with quantity

### ISSUE THREAD

**Issue by nesc58** [SonarSource/sonar-dotnet#5826]
### Description

Rule S2699 is false positive triggered using NSubstitute Received method with Quanity arg.

### Repro steps

```
public interface ITestClass
{
    public void DoSomething();
}

public class Class1
{
    [Test]
    public void Works() // All fine
    {
        var substitute = Substitute.For<ITestClass>();
        substitute.DoSomething();

        substitute.Received(1).DoSomething();
    }

    [Test]
    public void NotWorking() // S2699 is triggered
    {
        var substitute = Substitute.For<ITestClass>();
        substitute.DoSomething();
        substitute.DoSomething();

        substitute.Received(Quantity.Exactly(1)).DoSomething();
    }
}
```

### Expected behavior

This rule should not be triggeed using NSubstitutes Received with Quantity object.
Received with int comes from `NSubstitute.SubstituteExtensions`
Received with quantity comes from `NSubstitute.ReceivedExtensions.ReceivedExtensions`
Internally ReceivedExtensions.ReceivedExtensions is used by the SubstituteExtensions. 
The SubstituteExtensions provides a few shorthands
 `substitute.Received().DoSomething();`  ->  `substitute.Received(Quantity.AtLeastOne()).DoSomething();`
 `substitute.Received(1).DoSomething();`  ->  `substitute.Received(Quantity.Exactly(1)).DoSomething();`

The fix must be done here:
https://github.com/SonarSource/sonar-dotnet/blob/90b1585d5dd6f33184768973706b2901a82c2958/analyzers/src/SonarAnalyzer.CSharp/Rules/TestMethodShouldContainAssertion.cs#L42-L49
This should contain something like
```
{"Received", KnownType.NSubstitute_ReceivedExtensions_ReceivedExtensions},
{"ReceivedWithAnyArgs", KnownType.NSubstitute_ReceivedExtensions_ReceivedExtensions},
```

### Actual behavior

The rule is triggered when Received is used with Quantity object

### Related information

* MSBuild / dotnet version: 6.0.301
* Operating System: Windows 10
* SonarScanner MSBuild: 5.7.1
